components/rhs_thread/rhs_thread.tsx

```diff
@@ -32,40 +35,53 @@ export function renderThumbHorizontal() {
     return (<div/>);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: Props) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsThread extends React.Component {
-    static propTypes = {
-        posts: PropTypes.arrayOf(PropTypes.object).isRequired,
-        channel: PropTypes.object.isRequired,
-        selected: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.string,
-        currentUserId: PropTypes.string.isRequired,
-        previewCollapsed: PropTypes.string.isRequired,
-        previewEnabled: PropTypes.bool.isRequired,
-        socketConnectionStatus: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            removePost: PropTypes.func.isRequired,
-            selectPostCard: PropTypes.func.isRequired,
-            getPostThread: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    posts: Post[];
+    channel: Partial<Channel> | null;
+    selected: Record<string, any>;
+    previousRhsState?: string;
+    currentUserId: string;
+    previewCollapsed: string;
+    previewEnabled: boolean;
+    socketConnectionStatus: boolean;
+    actions: {
+        removePost: (post: ExtendedPost) => (dispatch: DispatchFunc, getState: GetStateFunc) => void;
```

Comment:
Is this type correct? `removePost` should've been wrapped by `bindActionCreators` so it won't return the inner function. I think it should probably be:
```suggestion
        removePost: (post: ExtendedPost) => void;
```